models/login_source.go

```diff
@@ -291,22 +297,37 @@ func GetLoginSourceByID(id int64) (*LoginSource, error) {
 	return source, nil
 }
 
+// ResetNonDefaultLoginSources clean other default source flag
+func ResetNonDefaultLoginSources(id int64) error {
+	_, err := x.NotIn("id", []int64{id}).Cols("is_default").Update(&LoginSource{IsDefault: false})
+	return err
+}
+
 // UpdateLoginSource updates information of login source to database or local file.
 func UpdateLoginSource(source *LoginSource) error {
 	if source.LocalFile == nil {
-		_, err := x.Id(source.ID).AllCols().Update(source)
-		return err
+		if _, err := x.Id(source.ID).AllCols().Update(source); err != nil {
```

Comment:
Why error occurs, usually means the update wasn't succeed, so `ResetNonDefaultLoginSources` at this time is not a good idea. I think we should only do `ResetNonDefaultLoginSources` if `err == nil`.

Reply:
Yes, that's a bug that should be addressed in the implementation, thanks I've included the fix in the PR.